RangeError: Maximum call stack size exceeded

on https://github.com/srghma/fix-github-https-repo-ps/tree/e00a40e5dffccb987831eaaef4f0e252ba131309

```sh
[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()" - --help
Installation complete.
Build succeeded.
/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107

[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()" - -h
Installation complete.
Build succeeded.
/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107

[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()" - --remote asdf
Installation complete.
Build succeeded.
/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107

[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()" - --unset-port
Installation complete.
Build succeeded.
/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107

[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()" - --port 123
Installation complete.
Build succeeded.
/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/fix-github-https-repo-ps/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/fix-github-https-repo-ps/output/Options.Applicative.Internal/index.js:396:107

[nix-shell:~/projects/fix-github-https-repo-ps]$ spago build && node -e "require('./output/Main/index.js').main()"
Installation complete.
Build succeeded.
{ remote: Nothing {}, port: Nothing {}, unset_port: false }
```

This error exists only on purescript 0.13, it doesnt throw on 0.12.5

I have reproduced it here https://github.com/f-o-a-m/purescript-optparse/pull/17

on purescript 0.12.5 and current master https://github.com/f-o-a-m/purescript-optparse/commit/782892c8d2015ab15c68ed2af46fb53b74169eaf

```
$ purs --version
0.12.5

$ rm -rf .pulp-cache output bower_components node_modules && npm i && bower i && npm run build && npm run test
... no errors

$ node -e "require('./output/Examples.Hello/index.js').main()" - -h
hello - a test for optparse-applicative

Usage: - --hello TARGET [-q|--quiet] [--repeat INT]
  Print a greeting for TARGET

Available options:
  --hello TARGET           Target for the greeting
  -q,--quiet               Whether to be quiet
  --repeat INT             Repeats for greeting (default: 1)
  -h,--help                Show this help text

```

on purescript .13 and https://github.com/f-o-a-m/purescript-optparse/commit/782892c8d2015ab15c68ed2af46fb53b74169eaf



```
$ purs --version
0.13.0
$ rm -rf .pulp-cache output bower_components node_modules && npm i && bower i 
$ npm run build

without errors
$ npm run test

Compiling Test.Main
           Src   Lib   All
Warnings   0     0     0
Errors     0     0     0
* Build successful.
* Running tests...
/home/srghma/projects/purescript-optparse/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/purescript-optparse/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
* ERROR: Subcommand terminated with exit code 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ test: `pulp test`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/srghma/.npm/_logs/2019-06-08T13_50_47_394Z-debug.log

$ node -e "require('./output/Examples.Hello/index.js').main()" - -h
/home/srghma/projects/purescript-optparse/output/Data.Functor/index.js:7
var Functor = function (map) {
                       ^

RangeError: Maximum call stack size exceeded
    at new Functor (/home/srghma/projects/purescript-optparse/output/Data.Functor/index.js:7:24)
    at listTFunctor (/home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:395:12)
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:83
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107
    at /home/srghma/projects/purescript-optparse/output/Options.Applicative.Internal/index.js:396:107


```

--- a/package.json
+++ b/package.json
@@ -5,8 +5,8 @@
     "test": "pulp test"
   },
   "devDependencies": {
-    "purescript-psa": "^0.6.0",
-    "purescript": "^0.12.0",
-    "pulp": "^12.2.0"
+    "purescript-psa": "^0.7.3",
+    "purescript": "^0.13.0",
+    "pulp": "^13.0.0"
   }
 }

--- a/src/Options/Applicative/Internal.purs
+++ b/src/Options/Applicative/Internal.purs
@@ -217,9 +217,9 @@ runListT xs = do
     TCons x xt -> liftM1 (List.Cons x) (runListT xt)
 
 instance listTFunctor :: Monad m => Functor (ListT m) where
-  map f = ListT
-         <<< liftM1 (bimapTStep f (map f))
-         <<< stepListT
+  map f v = ListT
+         $ liftM1 (bimapTStep f (map f))
+         $ stepListT v
 
 instance listTApply :: Monad m => Apply (ListT m) where
   apply = ap

